verbit/streaming_client.py

```diff
@@ -31,14 +33,25 @@ def from_name(cls, name: str):
         return cls.__members__.get(title)
 
 
-class WebSocketStreamingClient:
+class WebsocketStreamingClientSingleConnection:
 
     # constants
-    DEFAULT_CONNECT_TIMEOUT_SECONDS = 60.0
+    DEFAULT_CONNECT_TIMEOUT_SECONDS = 120.0
 
     # events
     EVENT_EOS = 'EOS'
 
+    # connection related exception classes:
+    #  1. WebSocketException: Raised from remote WebSocket connection
+    #  2. ConnectionError: Raised when WebSocket is on the same local machine
+    #  3. TimeoutError: From OS-level, for example when physically disconnecting from a network results in a timeout
+    #  4. socket.timeout/socket.gaierror: Are merged into '3' in python 3.10, but required in earlier python versions
+    CONNECTION_EXCEPTION_CLASSES = (WebSocketException, ConnectionError, TimeoutError, socket.timeout, socket.gaierror)
+
+    # retry related http status codes
+    RETRY_HTTP_4XX_CODES = (429,)
+    NO_RETRY_HTTP_5XX_CODES = (501, 505, 506, 507, 508, 510)
```

Comment:
I wouldn't commit to the code-ranges in the var names... 

```suggestion
    RETRY_HTTP_CODES = (429,)
    NO_RETRY_HTTP_CODES = (501, 505, 506, 507, 508, 510)
```

Reply:
I think it makes sense if you separate to different default retry behaviors for client/server errors.

Reply:
yes, maybe this is clearer in context, as these have opposite fallbacks:
 all other 4xx are "NO_RETRY"
 all other 5xx are "RETRY"
 might be clearer as local variables in the method

Reply:
We renamed to `RETRY_HTTP_CLIENT_CODES` and `NO_RETRY_HTTP_SERVER_CODES`
@roman-verbit-ai WDYT?

Reply:
Why do we care if it's client or server?
I thought we only care if to retry or not